### PR TITLE
add steps to migrate the projects to use the sdk version 0.12

### LINKED ISSUE #2161: v0.12.0 upgrade guide
## Type of question

About general context

## Question

Is there any v0.12.0 upgrade guide?

I didn't find any notes about in the relevant [documentation section](https://github.com/operator-framework/operator-sdk/blob/master/doc/migration/version-upgrade-guide.md), but I see breaking changes in release notes, e.g. Go 1.13 is required.

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
Add steps to migrate the projects to use SDK 0.12

**Motivation for the change:**

Closes #2161

### CODE REVIEW

File: doc/migration/version-upgrade-guide.md [hunk 1]
```diff
@@ -486,6 +486,75 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
...
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20190918202429-08c8357f8e2d
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20190918202713-c34a54b3ec8e
+	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20190918202550-958285cf3eef
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20190918203421-225f0541b3ea
+	k8s.io/metrics => k8s.io/metrics v0.0.0-20190918202012-3c1ca76f5bda
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190918201353-5cc279503896
+)
+```
+- Run `go mod tidy` to update the project modules
+- Run the command `operator-sdk generate k8s` to ensure that your resources will be updatedd
```
Comment (jmrodri) [inline doc/migration/version-upgrade-guide.md]: typo: updatedd should be updated

Comment (camilamacedo86) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
```

File: doc/migration/version-upgrade-guide.md [hunk 2]
```diff
@@ -486,6 +486,75 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
     - Delete the old CRD manifest files.
     - Rename CR manifest file names from `<group>_<version>_<kind>_cr.yaml` to `<full group>_<version>_<kind>_cr.yaml`.
 
+## `v0.12.x`
+
+### Go version
+
+- Ensure that you are using a go version 1.13+
+
+### modules
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: We need more detail about dependencies because we no longer support `dep`/`Gopkg.toml`. We should tell users how to [migrate to Go modules](https://blog.golang.org/migrating-to-go-modules) if they are still using `dep`.

File: doc/migration/version-upgrade-guide.md [hunk 3]
```diff
@@ -486,6 +486,75 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
...
+
+- Replace your `go.mod` file for:
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: It doesn't look like we have been telling users to do a full replacement of their dependency files, and for good reason, I think. Users may have made their own custom changes to bring in more dependencies. This instruction could cause their custom dependency inclusions to break or change in unexpected ways.

Rather, we should document which pieces of our `go.mod` scaffold changed between `v0.11.x` and `v0.12.x`.

File: doc/migration/version-upgrade-guide.md [hunk 4]
```diff
@@ -486,6 +486,75 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
...
+```go
+    ...
+    "github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/apis"
+	"github.com/operator-framework/operator-sdk-samples/memcached-operator/pkg/controller"
+	"github.com/operator-framework/operator-sdk-samples/memcached-operator/version" //This line
+    ...
+    func printVersion() {
+        log.Info(fmt.Sprintf("Operator Version: %s", version.Version)) //This line
+        log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+```
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: This snippet is confusing, I think, and spacing is a bit off.

How about the following (and note that I'm using tabs to conform to `go fmt`):
```go
import (
	...
	"<your_module_path>/version"
	...
)

func printVersion() {
	log.Info(fmt.Sprintf("Operator Version: %s", version.Version))
	...
}
```

File: doc/migration/version-upgrade-guide.md [hunk 5]
```diff
@@ -486,6 +486,75 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
...
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.0.0-20190918203421-225f0541b3ea
+	k8s.io/metrics => k8s.io/metrics v0.0.0-20190918202012-3c1ca76f5bda
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190918201353-5cc279503896
+)
+```
+- Run `go mod tidy` to update the project modules
+- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
+- Run the command `operator-sdk generate openapi` and ensure that no errors such as `API rule violation` are raised. For further information see the [API rules][api-rules] documentation. 
+
+### Main Go updates (Optional)
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
### (Optional) Update your operator to print its version
```

File: doc/migration/version-upgrade-guide.md [hunk 6]
```diff
@@ -486,6 +486,75 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
...
+
+The following change is to get the feature : Added `Operator Version: X.Y.Z` information in the operator logs.([#1953](https://github.com/operator-framework/operator-sdk/pull/1953))
+
+In the `<project>/cmd/manager/main.go` file, add the follow lines.
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
To add this feature to your operator, add the following lines in `<project>/cmd/manager/main.go`:
```

File: doc/migration/version-upgrade-guide.md [hunk 7]
```diff
@@ -486,6 +486,75 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
...
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190918201353-5cc279503896
+)
+```
+- Run `go mod tidy` to update the project modules
+- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
+- Run the command `operator-sdk generate openapi` and ensure that no errors such as `API rule violation` are raised. For further information see the [API rules][api-rules] documentation. 
+
+### Main Go updates (Optional)
+
+The following change is to get the feature : Added `Operator Version: X.Y.Z` information in the operator logs.([#1953](https://github.com/operator-framework/operator-sdk/pull/1953))
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
In v0.12.0, the SDK team updated the scaffold for `cmd/manager/main.go` to include the operator's version in the output produced by the `printVersion()` function. See [#1953](https://github.com/operator-framework/operator-sdk/pull/1953)
```

File: doc/migration/version-upgrade-guide.md [hunk 8]
```diff
@@ -486,6 +486,84 @@ All method signatures for [`sigs.k8s.io/controller-runtime/pkg/client.Client`](h
...
+        k8s.io/metrics => k8s.io/metrics v0.0.0-20190918202012-3c1ca76f5bda
+        k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.0.0-20190918201353-5cc279503896
+    )
+    ...
+```
+
+**NOTE**: Check [here](https://github.com/operator-framework/operator-sdk-samples/pull/90/files#diff-e15cac8b95d260726ca9db9fb25d9230) an example of this upgrade to see the changes from the version `0.11.0` to `0.12.0`.
+
+- Run `go mod tidy` to update the project modules
+- Run the command `operator-sdk generate k8s` to ensure that your resources will be updatedd
```
Comment (camilamacedo86) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
- Run the command `operator-sdk generate k8s` to ensure that your resources will be updated
```

**Comment by camilamacedo86** [comment]
HI @joelanford, 

Your comments and suggestions are addressed please, feel free to check it again.

File: doc/migration/version-upgrade-guide.md [hunk 9]
```diff
@@ -153,9 +153,9 @@ The following sections outline the upgrade steps for each SDK version along with
   deploy/olm-catalog/
   └── memcached-operator
       ├── 0.1.0
-      │   └── memcached-operator.v0.1.0.clusterserviceversion.yaml
+      │   └── memcached-operator.v0.1.0.clusterserviceversion.yaml
       ├── 0.2.0
-      │   └── memcached-operator.v0.2.0.clusterserviceversion.yaml
+      │   └── memcached-operator.v0.2.0.clusterserviceversion.yaml
```
Comment (joelanford) [inline doc/migration/version-upgrade-guide.md]: Nit: This change seems inadvertant/unrelated? Looking at the file with hexdump, it looks like `master` has two non-breaking spaces (`a0 20`) and your branch has two breaking spaces (`20`). Not sure if that matters, but since this is unrelated, I'd suggested reverting these lines.

File: doc/migration/version-upgrade-guide.md [hunk 10]
```diff
@@ -153,9 +153,9 @@ The following sections outline the upgrade steps for each SDK version along with
   deploy/olm-catalog/
   └── memcached-operator
       ├── 0.1.0
-      │   └── memcached-operator.v0.1.0.clusterserviceversion.yaml
+      │   └── memcached-operator.v0.1.0.clusterserviceversion.yaml
```
Comment (camilamacedo86) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
      │   └── memcached-operator.v0.1.0.clusterserviceversion.yaml
```

File: doc/migration/version-upgrade-guide.md (hunk 9 above)
Comment (camilamacedo86) [inline doc/migration/version-upgrade-guide.md]:
```suggestion
      │   └── memcached-operator.v0.2.0.clusterserviceversion.yaml
```
